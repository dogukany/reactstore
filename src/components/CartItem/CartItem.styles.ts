import { StyleSheet } from "react-native";

export default StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    padding: 10,
    borderBottomWidth: 1,
    borderBottomColor: "#eee",
  },
  image: {
    width: 100,
    height: 100,
    marginRight: 10,
  },
  title: {
    fontSize: 18,
    fontWeight: "bold",
  },
  price: {
    fontSize: 18,
    color: "#888",
  },
  deleteButton: {
    marginLeft: "auto",
    alignItems: "center",
    justifyContent: "center",
  },
  deleteText: {
    fontSize: 18,
    color: "red",
  },
  details: {
    alignItems: "flex-start",
    justifyContent: "center",
  },
});
