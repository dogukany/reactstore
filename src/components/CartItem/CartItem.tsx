import { Image, Text, View } from "react-native";
import type { IProduct } from "../../util/types/sliceTypes";
import styles from "./CartItem.styles";

interface CartItemProps {
  item: IProduct;
}

const CartItem = ({ item }: CartItemProps) => {
  return (
    <View style={styles.container}>
      <Image
        style={styles.image}
        source={{
          uri: item.imageUrl,
        }}
      />
      <View style={styles.details}>
        <Text style={styles.title}>{item.name}</Text>
        <Text style={styles.price}>${item.price}</Text>
      </View>
    </View>
  );
};

export default CartItem;
