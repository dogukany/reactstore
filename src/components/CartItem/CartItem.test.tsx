//renders correctly

import uuid from "react-native-uuid";
import renderer from "react-test-renderer";
import CartItem from "./CartItem";

describe("CartItem", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <CartItem
          item={{
            id: uuid.v4(),
            name: "Test",
            price: 100,
            imageUrl: "https://via.placeholder.com/150",
            description: "Test",
            shippingMethod: "spaceship",
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
