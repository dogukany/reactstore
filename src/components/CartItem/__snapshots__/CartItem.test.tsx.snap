// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem renders correctly 1`] = `
<View
  style={
    {
      "borderBottomColor": "#eee",
      "borderBottomWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "padding": 10,
    }
  }
>
  <Image
    source={
      {
        "uri": "https://via.placeholder.com/150",
      }
    }
    style={
      {
        "height": 100,
        "marginRight": 10,
        "width": 100,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "flex-start",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Test
    </Text>
    <Text
      style={
        {
          "color": "#888",
          "fontSize": 18,
        }
      }
    >
      $
      100
    </Text>
  </View>
</View>
`;
